shells/browser/shared/src/main.js

```diff
@@ -85,6 +85,16 @@ function createPanelIfReactLoaded() {
           );
         });
 
+        // Remember if we should sync the browser DevTools to the React tab.
+        // We'll only do that if user intentionally chooses a different React component.
+        let lastSelectedID = null;
+        bridge.addListener('selectElement', ({ id }) => {
+          if (!hasReactSelectionChanged && lastSelectedID !== id) {
```

Comment:
This looks wrong. We won't ever update `lastSelectedID` after the first time?

Reply:
Ah. I see what you're doing. I think the name of `lastSelectedID` just  confused me. It seemed like we should be updating it each time a new "selectElement" message was sent, but in this case we're just using it to decide when to set `hasReactSelectionChanged`.

I don't think we need it. Pretty sure we can just set `hasReactSelectionChanged = true` any time a "_selectElement_" message is sent. (We only send this message when a _new_ element is selected in the tree.)

```js
bridge.addListener('selectElement', ({ id }) => {
  hasReactSelectionChanged = true;
});
```

Reply:
That was my first attempt and it didn’t work because this message would get sent in other cases (such as when we sync DevTools *to* match the DOM node).

Reply:
Hmm...it seemed to work in my testing.

Reply:
It’s a bit tricky to test this specific piece. You’ll want to:

* select DOM node
* switch to Components
* switch back
* select a node that’s unrelated but doesn’t belong to another React component (like document head or some host sibling that’s still in the same React component)
* switch again to Components
* switch back again

The intended result is that DOM selection should stay on the host node you last chose. But with the approach you’re suggesting it would get reset again.

Reply:
Hmm. Actually I can't repro this anymore. Maybe I did something wrong when I tested. Your suggestion sounds good.